Dockerfile

```diff
@@ -1,8 +1,8 @@
-FROM ubuntu:20.04
-ARG VERSION=1.4.0
-ENV VERSION=${VERSION}
-RUN apt-get update && apt-get install -y wget
-RUN wget https://github.com/crate-ci/typos/releases/download/v${VERSION}/typos-v${VERSION}-x86_64-unknown-linux-musl.tar.gz && \
-    tar -xzvf typos-v${VERSION}-x86_64-unknown-linux-musl.tar.gz && \
-    mv typos /usr/local/bin
-ENTRYPOINT ["/usr/local/bin/typos"]
+FROM rust:1.58.1 as builder
+WORKDIR /usr/src/typos
+COPY . .
+RUN cargo install --path .
+
+FROM debian:buster-slim
+COPY --from=builder /usr/local/cargo/bin/typos /usr/local/bin/typos
```

Comment:
So this is updating our general docker image but the `docker/Dockerfile` is the one used by actions.  We use a bash script to get extra logic to integrate with the action.

I'm asuming we should keep that separate and have it be a layer on top of this image.

Reply:
okay good, I was trying to read up on it. 

I think that should stay until the core image is published and tagged, then yea, either merge the two and use image in the action.yml, or use the tagged version as FROM